### DISCUSSION TITLE
System Activity > Process: Handling Injection Attributes

### DISCUSSION THREAD

**Discussion by mikeradka** [ocsf/ocsf-schema#171] [Proposals]
Currently, [System Activity > Process Activity](https://schema.ocsf.io/classes/process_activity?extensions=) events have an `Inject` activity for injection events.

<img width="962" alt="image" src="https://user-images.githubusercontent.com/91983279/186256885-437d0ebf-b2eb-4185-96bf-8dae0e9c349d.png">

The [Mitre Process Injection Technique](https://attack.mitre.org/techniques/T1055/) defines Process Injection as 

_> "a method of executing arbitrary code in the address space of a separate live process."_

It was expressed that there is a need to accurately map properties of an injection - namely the `Injection Path`.

Currently, the [Process Activity Class](https://schema.ocsf.io/classes/process_activity?extensions=) has two injection attributes: `injection_type` and `injection_type_id`, but no discreet attribute for `Injection Path`:

<img width="1151" alt="image" src="https://user-images.githubusercontent.com/91983279/186258364-057344a6-ae1a-4157-b668-abc94c65c2cc.png">

Three approaches which came up in partner discussions are:

1. Keep the schema as is. In mapping implementation, `actor_process` injects `module` into the `target process`
2. Add an `injection_path` attribute to the Process Activity class (would mean one structural change - an addition)
3. Make an `injection` object with `.path`, `.type`, and `.type_id` attributes (would mean multiple structural changes, creating a new object and moving attributes into it)
4. Like 1. Keep the structure of the schema as is. In mapping implementation, `actor_process` injects `module` into the `target process`. In Addition, update the definition of `process` to "The process that was launched, injected into, opened, or terminated."

**Comment by mikeradka** [comment]
In any case, the definitions of `actor_process` and `process` were misleading and thus were updated in #172 

**Old Definitions:**

- `actor_process`: _The process that performed the operation or action on the target object. For example, the process that could have created a new file or violated a policy._
- `process`: _The process that was launched, injected, opened, or terminated._

**New Definitions:**

- `actor_process`: _The process that performed the operation or action on the target **`process`. For example, the process that could have started a new process or injected code into another process.**_
- `process`: _The process that was launched, injected **into**, opened, or terminated._

**Comment by JasonKeirstead** [comment]
Isn't a fifth option to change the type of injection_type to be an external reference that points at T1055 sub techniques? 

Why re-invent something already well defined and maintained.

**Reply by mikeradka** [reply]
I believe this one is being discussed in some of the subsequent discussions more specific to 'injection_type_id'.

**Comment by mikeradka** [comment]
Putting this one to bed, as we addressed the main issue of `injection_path` in the implementation of `module`.
